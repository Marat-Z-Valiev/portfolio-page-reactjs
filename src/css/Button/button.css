.button {
  background-color: #4594A4;
  margin-top: 20px;
  margin-bottom: 20px;
  margin-right: 5px;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .375rem .75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: .25rem;
  transition: background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
.button:hover {
  color: #fff;
  background-color: #0069d9;
  border-color: #0062cc;
  cursor: pointer;
}
.button span {
  vertical-align: 5px;
}
.button .fa {
  font-size: 3em;
  margin-right: 5px;
}
