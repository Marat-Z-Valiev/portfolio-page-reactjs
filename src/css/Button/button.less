.button{
    background-color: #4594A4;
    margin-top: 20px;
    margin-bottom: 20px;
    margin-right: 5px;
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 1px solid transparent;
    padding: .375rem .75rem;
    font-size: 1rem;
    line-height: 1.5;
    border-radius: .25rem;
    transition: background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;

    &:hover{
        color: #fff;
        background-color: #0069d9;
        border-color: #0062cc;
        cursor: pointer;
    }
    span {
    vertical-align: 5px;
  }

  .fa {
    font-size: 3em;
    margin-right: 5px;
  }
}