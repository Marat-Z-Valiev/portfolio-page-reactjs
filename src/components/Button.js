import React from "react";
import PropTypes from "prop-types";
import "../css/Button/button.css";

class Button extends React.Component {
	handleClick(props) {
		window.open(props, "_blank").opener = null;
	}
	render() {
		return (
			<button
				className="button"
				onClick={() => this.handleClick(this.props.link)}
			>
				<i className={this.props.icon} aria-hidden="true" />
				<span>{this.props.label}</span>
			</button>
		);
	}
}

Button.propTypes = {
	link: PropTypes.string,
	icon: PropTypes.string,
	label: PropTypes.string
};

export default Button;
