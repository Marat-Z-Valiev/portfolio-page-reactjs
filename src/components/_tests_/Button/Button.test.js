import React from "react";
import {shallow} from "enzyme";

import Button from "../../Button";

describe("Button", () => {
	const mockFn = jest.fn();
	it("should render correctly", () => {
		const component = shallow(<Button />);
		expect(component).toMatchSnapshot();
	});
	it("should have a click function", () => {
		window.open = jest.fn();
		const component = shallow(<Button />);
		component.simulate("click");
		expect(window.open).toHaveBeenCalled();
	});
});
