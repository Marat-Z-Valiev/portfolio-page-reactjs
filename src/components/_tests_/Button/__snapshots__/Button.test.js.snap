// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          aria-hidden="true"
        />,
        <span />,
      ],
      "className": "button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            aria-hidden="true"
          />,
          <span />,
        ],
        "className": "button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
